src/settings/Models/FinePolicy/FinePolicy.test.js

```diff
@@ -0,0 +1,122 @@
+import FinePolicy from './FinePolicy';
+import { OverdueFine, Metadata } from '../common';
+
+describe('FinePolicy', () => {
+  const basicPolicy = {
+    id: 'id',
+    name: 'name',
+    description: 'description',
+    countClosed: 0,
+    maxOverdueFine: '0.00',
+    forgiveOverdueFine: 1,
+    gracePeriodRecall: 5,
+    maxOverdueRecallFine: '5.00',
+  };
+
+  const policy = {
+    ...basicPolicy,
+    overdueFine: {},
+    overdueRecallFine: {
+      quantity: 1,
+      intervalId: 'id',
+    },
+  };
+
+  const finePolicyInstance = new FinePolicy(policy);
+
+  it('"FinePolicy" should have correct properties', () => {
+    expect(finePolicyInstance).toEqual(expect.objectContaining(basicPolicy));
+  });
+
+  it('"FinePolicy" should have correct properties if "policy" is not provided', () => {
+    const finePolicy = new FinePolicy();
+
+    expect(finePolicy).toEqual(expect.objectContaining({
+      id: undefined,
+      name: undefined,
+      description: undefined,
+      countClosed: undefined,
+      maxOverdueFine: undefined,
+      forgiveOverdueFine: undefined,
+      gracePeriodRecall: undefined,
+      maxOverdueRecallFine: undefined,
+    }));
+  });
+
+  it('"overdueFine" property should be instance of "OverdueFine"', () => {
+    expect(finePolicyInstance.overdueFine).toBeInstanceOf(OverdueFine);
+  });
+
+  it('"overdueRecallFine" property should be instance of "OverdueFine"', () => {
+    expect(finePolicyInstance.overdueRecallFine).toBeInstanceOf(OverdueFine);
+  });
+
+  it('"metadata" property should be instance of "Metadata"', () => {
+    expect(finePolicyInstance.metadata).toBeInstanceOf(Metadata);
+  });
+
+  it('"hasValue" should return true if required value more than 0', () => {
+    expect(finePolicyInstance.hasValue('maxOverdueRecallFine')).toEqual(true);
+  });
+
+  it('"hasValue" should return false if required value less than 0', () => {
+    expect(finePolicyInstance.hasValue('countClosed')).toEqual(false);
+  });
+
+  it('"hasNonZeroValue" should return true if it is possible to parse value', () => {
+    expect(finePolicyInstance.hasNonZeroValue('maxOverdueRecallFine')).toEqual(true);
+  });
+
+  it('"hasNonZeroValue" should return false if it is not possible to parse value', () => {
```

Comment:
Could you please add one more test to check parsable zero value?

Reply:
done